upload/common/uploaded_file.py

```diff
@@ -17,16 +17,39 @@ class UploadedFile:
 
     """
     The UploadedFile class represents newly-uploaded or previously uploaded files.
-
-    If the parameters to __init__() include 'name', 'data', 'content_type': a new S3 object will be created.
     """
 
+    @classmethod
+    def create(cls, upload_area, name=None, content_type=None, data=None):
+        s3object = upload_area.s3_object_for_file(name)
+        s3object.put(Body=data, ContentType=content_type)
+        return cls(upload_area, s3object=s3object)
+
     @classmethod
     def from_s3_key(cls, upload_area, s3_key):
         s3object = s3.Bucket(upload_area.bucket_name).Object(s3_key)
         return cls(upload_area, s3object=s3object)
 
-    def __init__(self, upload_area, name=None, content_type=None, data=None, s3object=None):
+    @classmethod
+    def from_db_id(cls, db_id):
+        db = UploadDB()
+        file_props = db.get_pg_record('file', record_id=db_id)
+        area_props = db.get_pg_record('upload_area', record_id=file_props['upload_area_id'])
+        from .upload_area import UploadArea
+        upload_area = UploadArea(area_props['uuid'])
+        s3object = upload_area.s3_object_for_file(file_props['name'])
+        return cls(upload_area, s3object=s3object)
+
+    def __init__(self, upload_area, s3object):
+        """
+        The object of init() is to:
+        - populate properties from the S3 object
+        - create a DB record for this file of one does not exist
```

Comment:
If __init__'s purpose is also this, why have a class create method?

Reply:
The purpose of `create` is to create a new S3 object.